sql/2025/07/row-block-allow-to-wrap-to-multiple-lines-usage.sql

```diff
@@ -0,0 +1,77 @@
+# HTTP Archive query to get % WordPress sites using Allow to wrap to multiple lines option in Row block.
+#
+# WPP Research, Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# See query results here: https://github.com/GoogleChromeLabs/wpp-research/pull/194
+
+DECLARE
+  DATE_TO_QUERY DATE DEFAULT '2025-07-01';
+
+CREATE TEMPORARY FUNCTION IS_CMS(technologies ARRAY<STRUCT<technology STRING, categories ARRAY<STRING>, info ARRAY<STRING>>>, cms STRING, version STRING) RETURNS BOOL AS (
+  EXISTS(
+    SELECT * FROM UNNEST(technologies) AS technology, UNNEST(technology.info) AS info
+    WHERE technology.technology = cms
+    AND (
+      version = ""
+      OR ENDS_WITH(version, ".x") AND (STARTS_WITH(info, RTRIM(version, "x")) OR info = RTRIM(version, ".x"))
+      OR info = version
+    )
+  )
+);
+
+WITH wordpress AS (
+  SELECT
+    client,
+    page,
+    BOOL(custom_metrics.cms.wordpress.block_theme) AS has_block_theme
+  FROM
+    `httparchive.crawl.pages`
+  WHERE
+    date = DATE_TO_QUERY
+    AND IS_CMS(technologies, 'WordPress', '')
+    AND is_root_page = TRUE
+),
+
+rowBlockDetection AS (
+  SELECT
+    client,
+    page,
+    TRUE AS has_row_block_use_wrap_option
+  FROM
+    `httparchive.crawl.requests`
+  WHERE
+    date = DATE_TO_QUERY
+    AND is_root_page = TRUE
+    AND is_main_document
+    AND REGEXP_CONTAINS(response_body, r'<div class="wp-block-group is-horizontal is-nowrap')
```

Comment:
The group block's tag name can be dynamic, it doesn't have to be a `<div>`. See https://github.com/WordPress/gutenberg/blob/8a5a63a4f284e11776a8d3881ec9b14a8965cb18/packages/block-library/src/group/edit.js#L35-L54

Also, is the list class names always deterministic like that? What if they're in a different order?

Reply:
Thanks for pointing it out i missed that.

Query updated.